### PR TITLE
Add support for offsetting position and angles

### PR DESCRIPTION
**Pull request by arthurdead**
and the model scale

### CODE REVIEW

File: scripting/viewmodel_override.sp [hunk 1]
```diff
@@ -20,13 +20,15 @@
 #pragma newdecls required
 
 #include <tf_custom_attributes>
+#include <cwx>
```
Comment (nosoop) [inline scripting/viewmodel_override.sp:23]: Mind migrating the implementation to only use the custom attribute system?  While the repo is intended to have attributes for CWX to use, I do still want them to be usable standalone.

`TF2CustAttr_GetAttributeKeyValues` is also available and subsections are allowed, so you should be able to implement it in a similar way; just sub the item UID check for checking that the subsections exist.

File: scripting/viewmodel_override.sp [hunk 2]
```diff
@@ -187,6 +189,71 @@ void UpdateClientWeaponModel(int client) {
...
 		bitsActiveModels |= MODEL_WORLD_ACTIVE;
 	}
+
+	if(bitsActiveModels & (MODEL_VIEW_ACTIVE|MODEL_WORLD_ACTIVE) != 0) {
+		char uid[64];
+		if(CWX_GetItemUIDFromEntity(weapon, uid, sizeof(uid)) && CWX_IsItemUIDValid(uid)) {
+			if(bitsActiveModels & MODEL_VIEW_ACTIVE != 0) {
+				KeyValues viewmodel_offset = CWX_GetItemExtData(uid, "viewmodel override offset");
+				if(viewmodel_offset) {
+					int weaponvm = EntRefToEntIndex(g_iLastViewmodelRef[client]);
```
Comment (nosoop) [inline scripting/viewmodel_override.sp:199]: You should be able to skip `EntRefToEntIndex` &mdash; everything here should be able to take the `g_iLastViewmodelRef[client]` entity reference directly, though you can do `int weaponvm = g_iLastViewmodelRef[client]` so you don't have to write the global out every time.

**Review by nosoop — CHANGES_REQUESTED** [review]
Nifty; I'd definitely want to see some examples of this in use.  Guessing they're for non-TF2 models?

File: scripting/viewmodel_override.sp (hunk 1 above)
Comment (arthurdead) [reply scripting/viewmodel_override.sp]: did i do it right?

**Comment by arthurdead** [comment]
> Nifty; I'd definitely want to see some examples of this in use. Guessing they're for non-TF2 models?

its just hl2 models

image

File: scripting/viewmodel_override.sp (hunk 1 above)
Comment (nosoop) [reply scripting/viewmodel_override.sp]: Should be fine, but I'll test it shortly and fixup some other minor nits.

**Comment by nosoop** [comment]
LGTM; thanks for the contribution!

**Pull request merged**